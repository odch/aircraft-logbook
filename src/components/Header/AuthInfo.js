import React from 'react'
import PropTypes from 'prop-types'

const AuthInfo = props => <div data-cy="username">{props.username}</div>

AuthInfo.propTypes = {
  username: PropTypes.string.isRequired
}

export default AuthInfo
