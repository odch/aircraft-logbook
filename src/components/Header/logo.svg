<?xml version="1.0" encoding="utf-8"?>
<!-- Generator: Adobe Illustrator 24.2.3, SVG Export Plug-In . SVG Version: 6.00 Build 0)  -->
<svg version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" x="0px" y="0px"
	 width="2000px" height="638px" viewBox="0 0 2000 638" style="enable-background:new 0 0 2000 638;" xml:space="preserve">
<style type="text/css">
	.st0{fill:#FFFFFF;}
	.st1{fill-rule:evenodd;clip-rule:evenodd;fill:#FFFFFF;}
	.st2{fill:#231F20;}
	.st3{fill-rule:evenodd;clip-rule:evenodd;fill:#231F20;}
	.st4{fill:#059FDB;}
	.st5{fill-rule:evenodd;clip-rule:evenodd;fill:#059FDB;}
</style>
<g>
	<g>
		<path class="st0" d="M471.8,118.7L310.6,490.2h-77.2L72.2,118.7h68.6L272.3,421l130.9-302.3H471.8z"/>
		<path class="st0" d="M517.5,337.6V118.7h62.9v218.9c0,53.2,45.2,96,100.6,96c55.4,0,100.6-42.9,100.6-96V118.7h62.9v218.9
			c0,87.4-73.7,158.9-163.5,158.9C590.6,496.5,517.5,425,517.5,337.6z"/>
		<path class="st0" d="M998.7,181.5v104h174.9v62.9H998.7v78.9H1205v62.9H935.8V118.7H1205v62.9H998.7z"/>
		<path class="st0" d="M1522.8,427.3v62.9h-226.3V118.7h62.9v308.6H1522.8z"/>
	</g>
</g>
<path class="st1" d="M1787,489.8l-28-375.7l168.8,375.7H1787z M1730.9,489.8h-140.7l168.8-375.7L1730.9,489.8z M1702.8,508.3h112.7
	l-56.1,37.6L1702.8,508.3z"/>
</svg>
