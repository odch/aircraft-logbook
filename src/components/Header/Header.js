import React from 'react'
import PropTypes from 'prop-types'
import AuthInfo from './AuthInfo'

class Header extends React.Component {
  render() {
    return (
      <React.Fragment>
        {!this.props.auth.isEmpty ? (
          <React.Fragment>
            <AuthInfo username={this.props.auth.email} />
            <button onClick={() => this.props.logout()}>Logout</button>
          </React.Fragment>
        ) : null}
      </React.Fragment>
    )
  }
}

Header.propTypes = {
  auth: PropTypes.shape({
    isEmpty: PropTypes.bool.isRequired,
    email: PropTypes.string
  }).isRequired,
  logout: PropTypes.func.isRequired
}

export default Header
