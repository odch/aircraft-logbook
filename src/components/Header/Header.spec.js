import React from 'react'
import renderer from 'react-test-renderer'
import Header from './Header'

describe('components', () => {
  describe('Header', () => {
    it('renders auth info if logged in', () => {
      const auth = {
        isEmpty: false,
        email: 'test@example.com'
      }
      const tree = renderer
        .create(<Header auth={auth} logout={() => {}} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('does not render if not logged in', () => {
      const auth = {
        isEmpty: true
      }
      const tree = renderer
        .create(<Header auth={auth} logout={() => {}} />)
        .toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('calls logout function', () => {
      const auth = {
        isEmpty: false,
        email: 'test@example.com'
      }

      const logout = jest.fn()

      const component = renderer.create(<Header auth={auth} logout={logout} />)

      const logoutButton = component.root.find(el => el.type === 'button')
      logoutButton.props.onClick()
      expect(logout).toHaveBeenCalled()
    })
  })
})
