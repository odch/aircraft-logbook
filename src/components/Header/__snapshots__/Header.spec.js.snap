// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Header does not render if not logged in 1`] = `null`;

exports[`components Header renders auth info if logged in 1`] = `
Array [
  <div
    data-cy="username"
  >
    test@example.com
  </div>,
  <button
    onClick={[Function]}
  >
    Logout
  </button>,
]
`;
